### PR TITLE
refactor self.selections per Qt's pattern

### LINKED ISSUE #201: self.selections should follow Qt's get/set pattern
When `self.selections` is used: https://github.com/BCDA-APS/gemviz/blob/4ceef1ee93b9d65bdec15c81592493d514529a78/gemviz/select_fields_tablemodel.py#L218

it is not handled with `setSelections()` and a `self._selections` internal attribute.  Refactor to be consistent with Qt's get/set pattern.  We use that everywhere else.  Or at least try to.

### PR DESCRIPTION
**Pull request by prjemian**
- close #201

### CODE REVIEW

**Comment by rodolakis** [comment]
@MDecarabas left that one to me, @prjemian I'll get to it later today.

File: gemviz/select_fields_tablemodel.py [hunk 1]
```diff
@@ -342,7 +341,20 @@ def setFields(self, fields):
             if field.selection is not None:
                 column_number = self.columnNumber(field.selection)
                 if column_number in self.checkboxColumns:
-                    self.selections[row] = field.selection
+                    self.setSelectionsItem(row, field.selection)
+
+    def selections(self, key):
+        """Pick the key from the plot selections dictionary."""
+        return self._selections[key]
```
[the remark this answers is not in the document]
Comment (prjemian) [reply gemviz/select_fields_tablemodel.py:348]: I tend to agree. It's a question of consistency (with Qt's pattern) or being concise. Not sure these changes help.

@rodolakis - What do you think?

[the remark this answers is not in the document]
Comment (rodolakis) [reply gemviz/select_fields_tablemodel.py:348]: I am looking to see if I am using this at some other places in the code in mdaviz (stuff that would come up afterward as the development progresses)

[the remark this answers is not in the document]
Comment (rodolakis) [reply gemviz/select_fields_tablemodel.py:348]: The `def selections(self)` method is still not consistent with Qt pattern: `self._selections` attribute is a dictionary, so `self.selections()` should also return a dictionary.

File: gemviz/select_fields_tablemodel.py [hunk 2]
```diff
@@ -342,7 +341,20 @@ def setFields(self, fields):
...
+        return self._selections
+
+    def setSelections(self, selections=None):
+        """Plot selections: dict(row_number=column number}"""
+        selections = selections or {}
+        self._selections = {k: v for k, v in selections.items() if v is not None}
+
+    def setSelectionsItem(self, key, value):
+        """Set the key in the plot selections dictionary."""
+        self.selections()[key] = value
```
Comment (rodolakis) [inline gemviz/select_fields_tablemodel.py:357]: We should not modify the method (`self.selections`) but the attribute (`self._selections`)

**Review by rodolakis — APPROVED** [review]
All my comment were addressed, LGTM now!

**Pull request merged**